src/Avalonia.Diagnostics/Diagnostics/ViewModels/ControlDetailsViewModel.cs

```diff
@@ -115,13 +116,14 @@ public ControlDetailsViewModel(TreePageViewModel treePage, IVisual control)
             }
         }
 
-        private (object resourceKey, bool isDynamic)? GetResourceInfo(object value)
+        private (object resourceKey, bool isDynamic)? GetResourceInfo(object? value)
         {
             if (value is StaticResourceExtension staticResource)
             {
                 return (staticResource.ResourceKey, false);
             }
-            else if (value is DynamicResourceExtension dynamicResource)
+            else if (value is DynamicResourceExtension dynamicResource
+                && dynamicResource.ResourceKey != null)
```

Comment:
It seems that static resource extension is not annotated since both static and dynamic might have `null` resource key.

Reply:
Yes, ideally core projects should have been annotated first. So we will need to have second pass on this project after.
Still this PR gives at least something.